### PR TITLE
Disable installers

### PR DESCRIPTION
**Pull request by timbussmann**
Allows to disable installers to address #4893

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Core/Installers/When_installer_behavior_not_specified.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                .Done(c => c.EndpointsStarted)
+                .Run();
+
+            Assert.IsFalse(context.InstallerCalled);
+        }
+
+        [Test]
+        public async Task Should_run_installers_when_debugger_attached()
+        {
+            if (!Debugger.IsAttached)
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/Installers/When_installer_behavior_not_specified.cs]: that's a bit shitty, but maybe this is better than not having this test? I'm on the fence.

Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Core/Installers/When_installer_behavior_not_specified.cs]: I think it adds value just as documentation. Also we're likely to kill that feature before we release anyway?

Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Core/Installers/When_installer_behavior_not_specified.cs]: Strike that, was thinking v7 (still think the test add value as doco?)

**Comment by andreasohlund** [comment]
Failing tests seems to be due to the netcore stuff on the buildserver, can we disable that for `<v7` builds?

File: src/NServiceBus.Core/InitializableEndpoint.cs [hunk 2]
```diff
@@ -160,22 +160,33 @@ static bool IsIProvideConfiguration(Type type)
...
+                container.ConfigureComponent(installerType, DependencyLifecycle.InstancePerCall);
             }
+
+            var username = GetInstallationUserName();
+            foreach (var installer in builder.BuildAll<INeedToInstallSomething>())
+            {
+                await installer.Install(username).ConfigureAwait(false);
+            }
+
+
```
Comment (andreasohlund) [inline src/NServiceBus.Core/InitializableEndpoint.cs]: Nitpick: 🔥 the whitespace

File: src/NServiceBus.Core/InitializableEndpoint.cs [hunk 3]
```diff
@@ -160,22 +160,33 @@ static bool IsIProvideConfiguration(Type type)
             return typeof(IProvideConfiguration<>).MakeGenericType(args)
                 .IsAssignableFrom(type);
         }
-
+    
         async Task RunInstallers(IEnumerable<Type> concreteTypes)
         {
-            if (Debugger.IsAttached || settings.GetOrDefault<bool>("Installers.Enable"))
+            var shouldRunInstaller = settings.GetOrDefault<bool?>("Installers.Enable");
```
Comment (andreasohlund) [inline src/NServiceBus.Core/InitializableEndpoint.cs]: Nice trick with the nullable 👍

**Comment by andreasohlund** [comment]
Just nothing that there needs to be a doco update as well?

File: src/NServiceBus.Core/InitializableEndpoint.cs [hunk 4]
```diff
@@ -160,22 +160,33 @@ static bool IsIProvideConfiguration(Type type)
             return typeof(IProvideConfiguration<>).MakeGenericType(args)
                 .IsAssignableFrom(type);
         }
-
+
```
Comment (bording) [inline src/NServiceBus.Core/InitializableEndpoint.cs]: 🔥

**Comment by bording** [comment]
Just a reminder that Squash and merge is the best choice for merging this one.

**Comment by timbussmann** [comment]
removed the whitespaces

**Pull request merged**